contracts/protocol/clients/voucher/BosonVoucher.sol

```diff
@@ -688,80 +743,76 @@ contract BosonVoucherBase is IBosonVoucher, BeaconClientBase, OwnableUpgradeable
-            IBosonExchangeHandler(protocolDiamond).commitToPreMintedOffer(payable(_to), offerId, _tokenId);
+            // Set committed status
+            _committed[_tokenId] = committed;
         } else if (_from != address(0) && _to != address(0) && _from != _to) {
             // Update the buyer associated with the voucher in the protocol
             // Only when transferring, not when minting or burning
             onVoucherTransferred(_tokenId, payable(_to));
         }
     }
 
-    /**
-     * @dev Determines if a token is pre-minted and committable via transfer hook
-     *
-     * Committable means:
-     * - does not yet have an owner
-     * - in a reserved range
-     * - has been pre-minted
-     * - has not been already burned
-     *
-     * @param _tokenId - the token id to check
-     * @return committable - whether the token is committable
-     * @return offerId - the associated offer id if committable
-     */
-    function getPreMintStatus(uint256 _tokenId) public view returns (bool committable, uint256 offerId) {
-        // Not committable if _committed already or if token has an owner
-        if (!_committed[_tokenId] && !_exists(_tokenId)) {
-            // If are reserved ranges, search them
-            uint256 length = _rangeOfferIds.length;
-            if (length > 0) {
-                // Binary search the ranges array
-                uint256 low = 0; // Lower bound of search (array index)
-                uint256 high = length; // Upper bound of search
-
-                while (low < high) {
-                    // Calculate the current midpoint and get the offer id
-                    uint256 mid = (high + low) / 2;
-                    uint256 currentOfferId = _rangeOfferIds[mid];
-
-                    // Get the range stored at the midpoint
-                    Range storage range = _rangeByOfferId[currentOfferId];
-
-                    // Get the beginning of the range once for reference
-                    uint256 start = range.start;
-
-                    if (start > _tokenId) {
-                        // Split low and search again if target too high
-                        high = mid;
-                    } else if (start + range.length - 1 >= _tokenId) {
-                        // Is token in target's reserved range?
-
-                        if (start + range.minted - 1 >= _tokenId && _tokenId > range.lastBurnedTokenId) {
-                            // Has it been pre-minted and not burned yet?
-                            committable = true;
-                            offerId = currentOfferId;
-                        }
-                        break; // Found!
-                    } else {
-                        // No? It may be in a higher range
-                        low = mid + 1;
-                    }
+    function isTokenCommittable(uint256 _tokenId) public view returns (bool committable, address owner) {
```

Comment:
is returning an `owner` really that beneficial? 
In all branches where `ownerOf(_tokenId)` is used, it could surely be used in the same way in function that invoked `isTokenCommittable` in the first place.
So the only point where we gain a little is where `range.owner` is since we already have a pointer to `range` initialized. But to get this "gain" we need to call "exists" which is another SLOAD, so I think that getting the rangeOwner directly from the storage whenever is needed is better overall.

Reply:
owner removed